shared/lib_pvshade.cpp

```diff
@@ -299,6 +278,50 @@ void selfshade_xs_horstr(bool landscape,
 	}
 }
 
+double ssSkyDiff::lookup(double tilt) {
+    char buf[124];
+    sprintf(buf, "%.3f", tilt);
+    if (SkyDiffTable.find(buf) != SkyDiffTable.end())
+        return SkyDiffTable[buf];
+    return compute(tilt);
+}
+
+double ssSkyDiff::compute(double tilt) {
+    if (gcr == 0)
+        throw std::runtime_error("ssSkyDiff::compute error: gcr required in initialization");
+    // sky diffuse reduction
+    double step = 1.0 / 1000.0;
+    double skydiff = 0.0;
+    double tand_stilt = tand(tilt);
+    double sind_stilt = sind(tilt);
+    double Asky = M_PI + M_PI / pow((1 + pow(tand_stilt, 2)), 0.5);
```

Comment:
looks like original function for Asky required tan of stilt in radians, not degrees?

Reply:
Original version used tan(stilt * DTOR)
```
Asky = M_PI + M_PI / pow((1 + pow(tan(stilt * DTOR), 2)), 0.5)
```
whereas now it uses 
```
#define tand(x) tan( DTOR*(x) )
...
tand_stilt = tand(surface_tilt)
```

The reason I reorganized this code is because I initially started by trying to speed up the simulation by vectorizing the calculations (hence all the 1000-length arrays) and this resulted in some speed up, but not enough to make a dent. So then I implemented the whole look-up table, and decided to leave this as is.